### PR TITLE
Run flooder

### PR DESCRIPTION
**Pull request by kostekIV**

### CODE REVIEW

File: benchmark/experimenting/utils.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import logging
+import os
+import stat
+
+from pathlib import Path
+from shutil import copyfile
+
+
+def copy_binary(binary: Path, target: str):
+    logging.info(f'Copying binary from {binary}...')
```
Comment (pmikolajczyk41) [inline benchmark/experimenting/utils.py:10]: `f'Copying binary from {binary} to {target}...'`

File: benchmark/benchmark.py [hunk 2]
```diff
@@ -30,6 +31,16 @@ def get_args() -> argparse.Namespace:
     parser_clean.add_argument('tag', type=str, help='tag for the machines', default='b2')
     parser_clean.set_defaults(func=clean)
 
+    parser_flooder = subparsers.add_parser('flooder')
```
Comment (pmikolajczyk41) [inline benchmark/benchmark.py:34]: for consistency `.add_parser('flood')`

File: benchmark/benchmark.py [hunk 3]
```diff
@@ -30,6 +31,16 @@ def get_args() -> argparse.Namespace:
...
+    parser_flooder.add_argument('--flooder-binary', type=Path, help='flooder executable')
+    group = parser_flooder.add_mutually_exclusive_group()
+    group.add_argument('--phrase', type=str, help='secret phrase : a path to a file or passed on stdin')
+    group.add_argument('--seed', type=str, help='secret seed of the account keypair passed on stdin')
+    parser_flooder.add_argument('--nodes', type=str, nargs='+', help='URL address(es) of the nodes to send transactions to')
+    parser_flooder.add_argument('--transactions', type=int, help='how many transactions to send')
```
Comment (pmikolajczyk41) [inline benchmark/benchmark.py:40]: for ease, I would add some short alias, like `--tx`

File: benchmark/benchmark.py [hunk 4]
```diff
@@ -30,6 +31,16 @@ def get_args() -> argparse.Namespace:
     parser_clean.add_argument('tag', type=str, help='tag for the machines', default='b2')
     parser_clean.set_defaults(func=clean)
 
+    parser_flooder = subparsers.add_parser('flooder')
+    parser_flooder.add_argument('--flooder-binary', type=Path, help='flooder executable')
+    group = parser_flooder.add_mutually_exclusive_group()
+    group.add_argument('--phrase', type=str, help='secret phrase : a path to a file or passed on stdin')
+    group.add_argument('--seed', type=str, help='secret seed of the account keypair passed on stdin')
+    parser_flooder.add_argument('--nodes', type=str, nargs='+', help='URL address(es) of the nodes to send transactions to')
```
Comment (pmikolajczyk41) [inline benchmark/benchmark.py:39]: I guess it is be sometimes easier to pass the `addresses` file, so I would enable also this way

File: benchmark/experimenting/flooder.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        nodes=args.nodes,
+        seed=args.seed,
+        phrase=args.phrase,
+        transactions=args.transactions,
+        throughput=args.throughput
+    )
+
+    setup_flooding()
+
+    terminate_instances_in_region(tag='flooder2')
```
Comment (pmikolajczyk41) [inline benchmark/experimenting/flooder.py:44]: leftover?

File: benchmark/README.md [hunk 6]
```diff
@@ -76,6 +76,34 @@ one with Prometheus and the second one with Grafana servers.
...
+                        what throughput to use (transactions/s)
+  --tag TAG             tag for the machines
+```
+
+For example:
+```bash
+./benchmark.py flooder --flooder-binary ../flooder/target/release/flooder --nodes "ips" --transactions=100 --phrase="..."
+```
+
+where ips are ips of the instances run by `run` subcommand.
```
Comment (pmikolajczyk41) [inline benchmark/README.md:105]: where `"ips"` are ...

Comment (kostekIV) [reply benchmark/README.md]: no longer applicable :)

File: benchmark/benchmark.py (hunk 4 above)
Comment (kostekIV) [reply benchmark/benchmark.py]: done

File: benchmark/benchmark.py (hunk 3 above)
Comment (kostekIV) [reply benchmark/benchmark.py]: done

File: benchmark/experimenting/utils.py (hunk 1 above)
Comment (kostekIV) [reply benchmark/experimenting/utils.py]: done

File: benchmark/benchmark.py (hunk 2 above)
Comment (kostekIV) [reply benchmark/benchmark.py]: ok

**Review by pmikolajczyk41 — APPROVED** [review]
seems ok, however, I would test this code after merging https://github.com/Cardinal-Cryptography/Aleph-Testnet/pull/18 before merging this PR

**Review by mike1729 — APPROVED** [review]

**Pull request merged**